diff --git a/package/sway/sway.mk b/package/sway/sway.mk
index 19f90b8b00..73d2174930 100644
--- a/package/sway/sway.mk
+++ b/package/sway/sway.mk
@@ -8,7 +8,7 @@ SWAY_VERSION = 1.8.1
 SWAY_SITE = https://github.com/swaywm/sway/releases/download/$(SWAY_VERSION)
 SWAY_LICENSE = MIT
 SWAY_LICENSE_FILES = LICENSE
-SWAY_DEPENDENCIES = systemd host-pkgconf wlroots json-c pcre cairo pango
+SWAY_DEPENDENCIES = host-pkgconf wlroots json-c pcre cairo pango
 SWAY_CONF_OPTS = \
 	-Dwerror=false \
 	-Ddefault-wallpaper=false \
@@ -18,8 +18,7 @@ SWAY_CONF_OPTS = \
 	-Dswaybar=false \
 	-Dswaynag=false \
 	-Dtray=disabled \
-	-Dman-pages=disabled \
-	-Dsd-bus-provider=libsystemd
+	-Dman-pages=disabled
 
 ifeq ($(BR2_PACKAGE_WLROOTS_X11),y)
 SWAY_CONF_OPTS += -Dxwayland=enabled
diff --git a/package/wlroots/wlroots.mk b/package/wlroots/wlroots.mk
index bb5c8f497d..20ed5d6b65 100644
--- a/package/wlroots/wlroots.mk
+++ b/package/wlroots/wlroots.mk
@@ -14,6 +14,8 @@ WLROOTS_DEPENDENCIES = \
 	host-pkgconf \
 	host-wayland \
 	hwdata \
+	libdrm \
+	libgbm \
 	libinput \
 	libxkbcommon \
 	libegl \
