From a6404820e5538870dc406fb9f155427d57c2d50c Mon Sep 17 00:00:00 2001
From: brrrrrrrrrr <brrrrrrrrrr@brrrrrrrrrr>
Date: Thu, 12 Oct 2023 23:06:30 -0400
Subject: [PATCH] rk3566 kernel copt optimizations

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index c7ee53f4bf04..6044557ffc2a 100644
--- a/Makefile
+++ b/Makefile
@@ -842,6 +842,7 @@ stackp-flags-$(CONFIG_STACKPROTECTOR)             := -fstack-protector
 stackp-flags-$(CONFIG_STACKPROTECTOR_STRONG)      := -fstack-protector-strong
 
 KBUILD_CFLAGS += $(stackp-flags-y)
+KBUILD_CFLAGS += -mtune=cortex-a55 -mcpu=cortex-a55+crc+crypto+fp+simd
 
 KBUILD_RUSTFLAGS-$(CONFIG_WERROR) += -Dwarnings
 KBUILD_RUSTFLAGS += $(KBUILD_RUSTFLAGS-y)
-- 
2.44.0

