From 6a9fea3254b84d5f82b0e96b8143e6036be522a3 Mon Sep 17 00:00:00 2001
From: brrrrrrrrrr <brrrrrrrrrr@brrrrrrrrrr>
Date: Sun, 21 Jan 2024 22:46:26 -0500
Subject: [PATCH 1/5] shutdown/reboot

---
 retroarch.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/retroarch.c b/retroarch.c
index ddc0ca9eda..bf736243c9 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -3465,7 +3465,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -P now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -P now");
+         system("poweroff");
 #endif /* HAVE_LAKKA */
 #endif
          break;
@@ -3480,7 +3480,7 @@ bool command_event(enum event_command cmd, void *data)
          system("(sleep 1 && shutdown -r now) & disown");
 #else
          command_event(CMD_EVENT_QUIT, NULL);
-         system("shutdown -r now");
+         system("reboot");
 #endif /* HAVE_LAKKA */
 #endif
          break;
-- 
2.43.0

